pkg/crypto/ed25519.go

```diff
@@ -0,0 +1,74 @@
+package crypto
+
+import (
+	"bytes"
+	"crypto/ed25519"
+	"encoding/hex"
+	"strings"
+)
+
+const (
+	ed25519Prefix = 0xED
+)
+
+type ed25519CryptoAlgorithm CryptoAlgorithm
+
+func ED25519() ed25519CryptoAlgorithm {
+	return ed25519CryptoAlgorithm{
+		prefix: ed25519Prefix,
+	}
+}
+
+func (c ed25519CryptoAlgorithm) Prefix() byte {
+	return c.prefix
+}
+
+func (c ed25519CryptoAlgorithm) FamilySeedPrefix() byte {
+	return c.familySeedPrefix
+}
+
+func (c ed25519CryptoAlgorithm) DeriveKeypair(decodedSeed []byte, validator bool) (string, string, error) {
```

Comment:
Do we also want another function that takes the seed (s...) as input like in `xrpl.js` ?

Reply:
I though about it, the `DecodeSeed` function should live in the `address-codec` package since it handles address logic, and the `crypto` module is not related to xrpl logic. It is a generic package.